crates/red_knot_python_semantic/resources/mdtest/call/function.md

```diff
@@ -64,3 +64,196 @@ def _(flag: bool):
     # error: [possibly-unresolved-reference]
     reveal_type(foo())  # revealed: int
 ```
+
+## Wrong argument type
+
+### Positional argument, positional-or-keyword parameter
+
+```py
+def f(x: int) -> int:
+    return 1
+
+# error: 15 [invalid-argument-type] "Object of type `Literal["foo"]` cannot be assigned to parameter `x` of function `f`; expected type `int`"
+reveal_type(f("foo"))  # revealed: int
+```
+
+### Positional argument, positional-only parameter
+
+```py
+def f(x: int, /) -> int:
+    return 1
+
+# error: 15 [invalid-argument-type] "Object of type `Literal["foo"]` cannot be assigned to parameter 0 (`x`) of function `f`; expected type `int`"
```

Comment:
I think we should use 1-based indexing for the parameter number -- computers think of numbers as starting at 0, but that's not how humans think, and the error message is for humans

Reply:
Done. I don't think there's any need for `OneIndexed` here, though -- I think it's less confusing if we continue to use exclusively zero-based indices in the code, and just `+ 1` at the display layer. At least for now; maybe this will change as the diagnostic system gets more sophisticated.

Reply:
b2e3bf1ae58ea9034d112a3cfd06fc85e56a207f